sycl/include/CL/sycl/ONEAPI/intel_matrix/matrix-amx.hpp

```diff
@@ -0,0 +1,439 @@
+//==------------- submatrix-amx.hpp - SYCL matrix -----------*- C++ -*---==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// ===--------------------------------------------------------------------=== //
+///
+/// We provide new interfaces for matrix muliply in this patch:
```

Comment:
As new interfaces are defined in `sycl::ext::intel` namespace, we probably should put this header to `sycl/include/CL/sycl/INTEL` directory instead of `sycl/include/CL/sycl/ONEAPI`.